joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1530,6 +1532,26 @@ bool JointTrajectoryController::has_active_trajectory() const
   return traj_external_point_ptr_ != nullptr && traj_external_point_ptr_->has_trajectory_msg();
 }
 
+void JointTrajectoryController::update_pids()
+{
+  for (size_t i = 0; i < dof_; ++i)
+  {
+    const auto & gains = params_.gains.joints_map.at(params_.joints[i]);
+    if (pids_.at(i))
+    {
+      // update PIDs with gains from ROS parameters
+      pids_[i]->setGains(gains.p, gains.i, gains.d, gains.i_clamp, -gains.i_clamp);
+    }
```

Comment:
Here we don't want to update `ff_velocity_scale` right?. I think it makes sense. However, just wanted to confirm😅

Reply:
`ff_velocity_scale` will be set in every case. I just had to differentiate if a new instance of PID has to be created, or just its parameters have to be set.

Reply:
Ahh yes, you are right. I missed the ending bracket. Thanks